Github/service.js

```diff
@@ -13,206 +13,124 @@ var request = require("request");
-                        fs.appendFileSync('output.txt', "DEBUG getTheRecentCommits added branchCommits[" + j + "] repo-branch: " + repoName + "-" + branchName + '\n');
-                        branchCommits[j].commit.repoName = repoName;
-                        branchCommits[j].commit.branchName = branchName;
-                        recentCommits.push(branchCommits[j].commit);
-                        /*recentCommits.push ({
-                            'repoName': repoName,
-                            'branchName': branchName,
-                            'commit': branchCommits[j].commit
-                        });*/
-
-                    } else {
-                        j = branchCommits.length;
-                    }
-                }
-                fs.appendFileSync('output.txt', '\n');
-                resolve();
-              }
+                let listOfBranches = {
+                    repo: branchObject.name,
+                    branches: []
+                };
+                JSON.parse(data).forEach(singleBranch => {
+                    singleBranch.repo = branchObject.name;
+                    listOfBranches.branches.push({name: singleBranch.name, sha: singleBranch.commit.sha});
+                });
+                resolve(listOfBranches);
+            }
         });
     });
-} 
+};
 
-// 4. sortRecentCommits
-module.exports.sortRecentCommits = () => {
+module.exports.getAllCommitsTogether = () => {
     return new Promise((resolve, reject) => {
-        recentCommits.sort((a,b) => {
-            let c = new Date(a.committer.date);
-            let d = new Date(b.committer.date);
-            return d - c;
+        let promises = [];
+        branchUrls.forEach((branchUrl) => {
+            promises.push(getBranches(branchUrl));
         });
-        fs.appendFileSync('outputArrays.txt', "sortRecentCommits sorted recent commits: " + recentCommits.length + "\n");
-        for (let i = 0; i < recentCommits.length; i++) {
-            fs.appendFileSync('outputArrays.txt', recentCommits[i].repoName + ":" + recentCommits[i].branchName + " - " + recentCommits[i].message + "\n");
-        }
-        resolve(recentCommits);
-    });
-}
-
-module.exports.delay = (t, v) => {
-    return new Promise((resolve) => {
-        setTimeout(resolve.bind(null, v), t);
+        Promise.all(promises)
+            .then((branchesPerRepo) => {
+                promises = [];
+                branchesPerRepo.forEach(rep =>{
+                    rep.branches.forEach(branch =>{
+                        promises.push(getCommits({repo: rep.repo, br: branch}));
+                    });
+                });
+                Promise.all(promises)
+                .then(()=>{
+                    recentCommits.sort((a, b) => {
+                        let c = new Date(a.commit.author.date);
+                        let d = new Date(b.commit.author.date);
+                        return d - c;
```

Comment:
`a` & `c` could be `const`? Also maybe update variable names `a`, `b`, `c`, `d` for clarity

Reply:
Sorry, I was just reusing code

Reply:
variable names changed